src/main/scripts/src/app/shared/perspectives/collections/post-it/post-it-collections-perspective.component.ts

```diff
@@ -18,23 +18,87 @@
  * -----------------------------------------------------------------------/
  */
 
-import {Component, Input} from '@angular/core';
+import {Component, Input, OnInit} from '@angular/core';
+import {trigger, state, style, transition, animate, keyframes} from '@angular/animations';
+
 import {CollectionService} from '../../../../core/rest/collection.service';
 import {Perspective} from '../../perspective';
+import {Collection} from '../../../../core/dto/collection';
+import {LocalService} from '../../../../core/local.service';
 
 @Component({
   selector: 'post-it-collections-perspective',
-  templateUrl: './post-it-collections-perspective.component.html'
+  templateUrl: './post-it-collections-perspective.component.html',
+  styleUrls: ['./post-it-collections-perspective.component.scss'],
+  animations: [
+    trigger('animateVisible', [
+      state('in', style({height: '*', width: '*', opacity: 1})),
+      transition('void => *', [
+        animate(200, keyframes([
+          style({height: 0, width: 0, opacity: 0, offset: 0}),
+          style({height: '*', width: '*', opacity: 1, offset: 1})
+        ]))
+      ]),
+      transition('* => void', [
+        animate(200, keyframes([
+          style({height: '*', width: '*', opacity: 1, offset: 0}),
+          style({height: 0, width: 0, opacity: 0, offset: 1})
+        ]))
+      ])
+    ])
+  ]
 })
-export class PostItCollectionsPerspectiveComponent implements Perspective {
+export class PostItCollectionsPerspectiveComponent implements Perspective, OnInit {
 
   @Input()
   public query: string;
 
   @Input()
   public editable: boolean;
 
-  constructor(private collectionService: CollectionService) {
+  public placeholderTitle: string = 'Collection name';
+  public iconsPerPage: number = 36;
+  public newCollection: Collection = new Collection();
+  public collections: Collection[];
+  public pickerVisible: boolean = false;
+  public icons: string[];
+  public colors: string[];
+  public numbers: any[];
+
+  constructor(private collectionService: CollectionService,
+              private localService: LocalService) {
+    this.numbers = Array.apply(null, {length: this.iconsPerPage}).map(Number.call, Number);
   }
 
+  public ngOnInit(): void {
+    this.collectionService.getCollections()
+      .subscribe((collections: Collection[]) => {
+        this.collections = collections;
+      });
+    this.localService.getIcons().subscribe((icons: string[]) => {
+      this.icons = icons;
+    });
+    this.localService.getColors().subscribe((colors: string[]) => {
+      this.colors = colors;
+    });
+  }
+
+  public saveCollection() {
+    this.collectionService.saveCollection(this.newCollection)
+      .subscribe((code: String) => {
+        this.newCollection = new Collection();
+        this.collectionService.getCollections()
+          .subscribe((collections: Collection[]) => {
```

Comment:
Subscribe does not work like that. Whenever you subscribe to service, the service is fired and function is executed. If we want to have lazy loaded collections we have 2 options (angular's way, or Rx.js way), both are valid.
1) Angular's way:
We'd have to do something like this ` this.collections = this.collectionService.getCollections()` and later on in template (notice the `async` pipe in `*ngFor`).
```html
<div class="col-sm-3 post-it" *ngFor="let oneCollection of collections | async">
  <span>Documents: {{oneCollection.documentCount}}</span>
  <div class="info">
    <i class="fa fa-2x" [ngClass]="oneCollection.icon" [ngStyle]="{'color':oneCollection.color}"></i>
    <h4>{{oneCollection.name}}</h4>
  </div>
</div>
```

2) Rx.js way
Using [behaviorsubject](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/subjects/behaviorsubject.md) which reacts to every change of promis, example can be seen at [behaviorsubject at scotch](https://scotch.io/tutorials/3-ways-to-pass-async-data-to-angular-2-child-components#Scotch.io_970x90_728x90_320x50_ATF_InPage3)